src/tree.service.ts

```diff
@@ -98,4 +107,18 @@ export class TreeService {
   public hasController(id: string | number): boolean {
     return this.controllers.has(id);
   }
+
+  private shouldFireLoadNextLevel(tree: Tree): boolean {
+
+    const result = tree.node.hasChildren &&
+      !tree.node.loadChildren &&
+      !tree.childrenAreBeingLoaded() &&
+      (!tree.children || tree.children === []);
+
+      if (result) {
```

Comment:
Probably instead of `result` it might be a bit more convenient to name it smth like `isChildrenLoadingRequested`